src/boot/boot.c

```diff
@@ -1776,11 +1782,35 @@ static size_t config_find_entry(Config *config, const char16_t *pattern) {
         return IDX_INVALID;
 }
 
+static bool config_sysfail_occured(Config *config) {
+        SysFailType sysfail_type;
+
+        assert(config);
+
+        sysfail_type = sysfail_check();
+        if (sysfail_type != SYSFAIL_OK) {
```

Comment:
Please invert the condition and return earlier.

Reply:
fixed